rules/typescript-enums/rule.md

```diff
@@ -0,0 +1,97 @@
+---
+type: rule
+title: Do you know why to use const assertions not TypeScript enums?
+uri: typescript-enums
+authors:
+  - title: Harry Ross
+    url: https://ssw.com.au/people/harry-ross
+related: 
+  - use-enums-instead-of-hard-coded-strings
+created: 2024-03-19T21:39:38.906Z
+archivedreason: null
+guid: ba19be99-354d-44b2-a2da-4131cc660f18
+---
+While it's super important to ensure that magic strings are not used in your codebase, there are better alternatives to using enums in TypeScript. You might expect TypeScript enums to function like strongly typed languages like C# and Rust, often this is not the case.

```

Comment:
```suggestion
As a rule, magic strings should be avoided at all costs because you don't want your code to rely on hardcoded string literals. The common solution is to use enums - which assign those strings to reusable variables. As this is usually the recommended solution for most strongly typed coding languages such as C# and Rust, it should be avoided when it comes to Typescript.

```